### DISCUSSION TITLE
Changing the client connection and disconnection method

### DISCUSSION THREAD

**Discussion by KeisukeYamashita** [prisma/prisma#13450] [Q&A]
## WHY

I want to use my Node.js app to connect to the Cloud Spanner using the [PostgreSQL interface](https://cloud.google.com/spanner/docs/postgresql-interface). I understand that the interface is not 100% compatible but I want to PoC for my toy app.

As far as I read the docs, there are no way to control the connection method. In many cases, the Postgres is connected by TCP but for Cloud Spanner, the the communication is on the gRPC protocol.

Is there a way to control this and use the Prisma's feature? 
Thank you in advance.

## What I want to do 

```javascript
# Connect to the Cloud Spanner gRPC API with my custom function
prisma.connect = async () => {}

# Run the query just like for other databases
const  users = await prisma.users.findMany({...})
```

Also, is there a way to developer the connectors? Thank you 🙏

**Comment by janpio** [comment]
> Is there a way to control this and use the Prisma's feature?

No. Unfortunately not.

**Reply by KeisukeYamashita** [reply]
Hi, thank you for your reply!
Just to be sure, am I correct that Prisma supports Postgres and SQLite, but not Cloud Spanner (Postgres interface), Cloudflare D1(SQLite interface), etc., which have different connection methods(e.g. driver)?

**Reply by janpio** [reply]
Generally correct:
- Cloud Spanner : https://github.com/prisma/prisma/issues/717 
- Cloudflare D1: https://github.com/prisma/prisma/issues/13310

If there is a way to make these happen though, please leave comments in those issues. You are not alone in wanting to be able to do that.

**Reply by KeisukeYamashita** [reply]
I didn't notice those issues, thank you very much!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
